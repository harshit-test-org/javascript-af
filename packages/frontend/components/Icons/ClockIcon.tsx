import React from 'react';

export const ClockIcon: React.FC<
  React.SVGAttributes<SVGSVGElement>
> = props => (
  <svg
    width={24}
    height={24}
    viewBox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    strokeLinecap="round"
    strokeLinejoin="round"
    role="img"
    {...props}
  >
    <title>Clock Icon</title>
    <circle cx="12" cy="12" r="10" />
    <polyline points="12 6 12 12 16 14" />
  </svg>
);
