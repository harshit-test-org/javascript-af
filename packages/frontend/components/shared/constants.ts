export const SANS_FAMILY = `-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif`;
export const MONO_FAMILY = ` 'Inconsolata', sans-serif`;
