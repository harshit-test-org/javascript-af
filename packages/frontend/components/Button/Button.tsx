import React from 'react';
import styled from '../../lib/styled-components';
import {
  color,
  space,
  ColorProps,
  SpaceProps,
  FontFamilyProps,
  fontFamily,
  fontWeight,
  FontWeightProps,
  BorderRadiusProps,
  borderRadius,
  fontSize,
  FontSizeProps,
  variant,
} from 'styled-system';
import Ink from 'react-ink';
import { SANS_FAMILY } from '../shared';

const applyVariant = variant({
  key: 'buttons',
});

const StyledButton = styled.button<
  ColorProps &
    SpaceProps &
    FontWeightProps &
    BorderRadiusProps &
    FontSizeProps & { variant: string } & React.AnchorHTMLAttributes<
      HTMLButtonElement | HTMLAnchorElement
    >
>`
  position: relative;
  border: 0;
  background: transparent;
  cursor: pointer;
  font-family: ${SANS_FAMILY};
  padding: 16px;
  margin: 8px;
  font-size: 1.2rem;
  border-radius: 44px;
  ${color}
  ${space}
  ${fontFamily}
  ${fontSize}
  ${borderRadius}
  ${applyVariant}
  ${fontWeight}
`;

export interface IButtonProps
  extends SpaceProps,
    FontFamilyProps,
    BorderRadiusProps,
    FontWeightProps,
    FontSizeProps,
    React.AnchorHTMLAttributes<HTMLButtonElement | HTMLAnchorElement> {
  variant?: string;
  as?: any;
}

export const Button: React.FC<IButtonProps> = ({
  children,
  variant: v = 'primary',
  ...others
}) => {
  return (
    <StyledButton role="button" variant={v} {...others}>
      {children}
      <Ink background={false} duration={700} />
    </StyledButton>
  );
};
