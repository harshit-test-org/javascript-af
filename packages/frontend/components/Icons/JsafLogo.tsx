import React from 'react';
import { theme } from '../shared';

export const JsafLogo: React.FC<React.SVGAttributes<SVGSVGElement>> = props => (
  <svg width={60} height={38} fill="none" role="img" {...props}>
    <title>Javascript AF logo</title>
    <path
      d="M10.714 25.577V2.923L15 0v25.577C15 36.538 8.571 38 0 38l2.857-4.385c5 0 7.857-.73 7.857-8.038z"
      fill="#fff"
    />
    <path d="M18 20L34 1l-5.5 15.5H38L25 37l3.5-17H18z" fill="#FFC600" />
    <path
      d="M35.284 35.198c-.288 0-.534-.102-.738-.306a.985.985 0 0 1-.306-.72c0-.276.102-.516.306-.72.204-.204.45-.306.738-.306.276 0 .516.102.72.306a.985.985 0 0 1 .306.72.985.985 0 0 1-.306.72.985.985 0 0 1-.72.306zm9.108-8.604c1.176 0 2.07.306 2.682.918.624.612.936 1.65.936 3.114V35h-1.404v-1.26c-.504.972-1.44 1.458-2.808 1.458-.912 0-1.626-.204-2.142-.612-.516-.42-.774-.972-.774-1.656 0-.576.186-1.08.558-1.512.372-.432.87-.762 1.494-.99a5.869 5.869 0 0 1 2.088-.36c.684 0 1.302.036 1.854.108-.072-.924-.318-1.572-.738-1.944-.408-.384-1.02-.576-1.836-.576-.408 0-.804.078-1.188.234a3.338 3.338 0 0 0-1.026.684l-.63-.828c.768-.768 1.746-1.152 2.934-1.152zm-.468 7.614c.888 0 1.59-.258 2.106-.774.516-.528.798-1.284.846-2.268a11.376 11.376 0 0 0-1.746-.126c-.912 0-1.632.15-2.16.45-.528.3-.792.756-.792 1.368 0 .9.582 1.35 1.746 1.35zm7.992-6.408h-1.782v-1.026h1.782v-.684c0-1.056.27-1.848.81-2.376.552-.528 1.29-.792 2.214-.792.528 0 1.014.096 1.458.288.456.192.828.474 1.116.846l-.684.954-.108.144-.108-.072a.59.59 0 0 0-.036-.144.71.71 0 0 0-.09-.252 1.01 1.01 0 0 0-.216-.252 1.699 1.699 0 0 0-.558-.36 2.13 2.13 0 0 0-.738-.108c-.564 0-1.008.168-1.332.504-.324.336-.486.876-.486 1.62v.684h2.7V27.8h-2.7V35h-1.242v-7.2z"
      fill="#D7D1D1"
    />
  </svg>
);

export const JsafLogoBlack: React.FC<
  React.SVGAttributes<SVGSVGElement>
> = props => (
  <svg width={60} height={38} fill="none" role="img" {...props}>
    <title>Javascript AF logo</title>
    <path
      d="M10.714 25.577V2.923L15 0v25.577C15 36.538 8.571 38 0 38l2.857-4.385c5 0 7.857-.73 7.857-8.038z"
      fill={theme.colors.primary.main}
    />
    <path d="M18 20L34 1l-5.5 15.5H38L25 37l3.5-17H18z" fill="#FFC600" />
    <path
      d="M35.284 35.198c-.288 0-.534-.102-.738-.306a.985.985 0 0 1-.306-.72c0-.276.102-.516.306-.72.204-.204.45-.306.738-.306.276 0 .516.102.72.306a.985.985 0 0 1 .306.72.985.985 0 0 1-.306.72.985.985 0 0 1-.72.306zm9.108-8.604c1.176 0 2.07.306 2.682.918.624.612.936 1.65.936 3.114V35h-1.404v-1.26c-.504.972-1.44 1.458-2.808 1.458-.912 0-1.626-.204-2.142-.612-.516-.42-.774-.972-.774-1.656 0-.576.186-1.08.558-1.512.372-.432.87-.762 1.494-.99a5.869 5.869 0 0 1 2.088-.36c.684 0 1.302.036 1.854.108-.072-.924-.318-1.572-.738-1.944-.408-.384-1.02-.576-1.836-.576-.408 0-.804.078-1.188.234a3.338 3.338 0 0 0-1.026.684l-.63-.828c.768-.768 1.746-1.152 2.934-1.152zm-.468 7.614c.888 0 1.59-.258 2.106-.774.516-.528.798-1.284.846-2.268a11.376 11.376 0 0 0-1.746-.126c-.912 0-1.632.15-2.16.45-.528.3-.792.756-.792 1.368 0 .9.582 1.35 1.746 1.35zm7.992-6.408h-1.782v-1.026h1.782v-.684c0-1.056.27-1.848.81-2.376.552-.528 1.29-.792 2.214-.792.528 0 1.014.096 1.458.288.456.192.828.474 1.116.846l-.684.954-.108.144-.108-.072a.59.59 0 0 0-.036-.144.71.71 0 0 0-.09-.252 1.01 1.01 0 0 0-.216-.252 1.699 1.699 0 0 0-.558-.36 2.13 2.13 0 0 0-.738-.108c-.564 0-1.008.168-1.332.504-.324.336-.486.876-.486 1.62v.684h2.7V27.8h-2.7V35h-1.242v-7.2z"
      fill="#000"
    />
  </svg>
);
