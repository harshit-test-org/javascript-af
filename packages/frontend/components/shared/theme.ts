import { transitions } from 'polished';
import { SANS_FAMILY, MONO_FAMILY } from './constants';

type swatch =
  | 'darkest'
  | 'darker'
  | 'dark'
  | 'main'
  | 'light'
  | 'lighter'
  | 'lightest';

export interface ITheme {
  colors: {
    primary: Record<swatch, string>;
    white: string;
    offWhite: string;
    black: string;
    blue: string;
  };
  fontFamily: {
    sans: string;
    mono: string;
  };
  buttons: {
    tabs: {
      backgroundColor: string;
      borderRadius: number;
      '&.active': {
        boxShadow: string;
      };
    };
    primary: {
      color: string;
      backgroundColor: string;
      '&:hover': {
        opacity: string;
        backgroundColor: string;
      };
    };
    icon: {
      padding: string;
      display: string;
      alignItems: string;
      '&:hover': {
        opacity: string;
      };
    };
    text: {
      padding: string;
      display: string;
      alignItems: string;
      '&:hover': {
        opacity: string;
      };
    };
    outlined: {
      backgroundColor: string;
      borderRadius: string;
      border: string;
      color: string;
      '&:hover': {
        opacity: string;
      };
    };
  };
  maxWidth: string;
}

const primarySwatch: Record<swatch, string> = {
  darkest: '#2B0448',
  darker: '#300958',
  dark: '#38116E',
  main: '#452984',
  light: '#5D48A7',
  lighter: '#7568C5',
  lightest: '#A098F1',
};

const theme: ITheme = {
  colors: {
    primary: primarySwatch,
    white: '#FFFFFF',
    offWhite: '#fefefe',
    black: '#000000',
    blue: '#2F47C7',
  },
  fontFamily: {
    sans: SANS_FAMILY,
    mono: MONO_FAMILY,
  },
  buttons: {
    tabs: {
      backgroundColor: '#fff',
      borderRadius: 0,
      '&.active': {
        boxShadow: `${primarySwatch.main} 0px -4px 0px inset`,
      },
    },
    primary: {
      color: 'white',
      backgroundColor: primarySwatch.main,
      ...transitions('0.2s background ease'),
      '&:hover': {
        opacity: '0.9',
        backgroundColor: primarySwatch.light,
      },
    },
    icon: {
      padding: '8px',
      display: 'flex',
      alignItems: 'center',
      '&:hover': {
        opacity: '0.9',
      },
    },
    text: {
      padding: '8px',
      display: 'inline-block',
      alignItems: 'center',
      '&:hover': {
        opacity: '0.9',
      },
    },
    outlined: {
      backgroundColor: 'white',
      borderRadius: '44px',
      border: `2px solid ${primarySwatch.main}`,
      color: primarySwatch.main,
      '&:hover': {
        opacity: '0.9',
      },
    },
  },
  maxWidth: '1370px',
};

export { theme as default };
