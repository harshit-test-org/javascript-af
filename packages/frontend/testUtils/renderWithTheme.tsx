import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from '../lib/styled-components';
import { theme } from '../components/shared';

export function renderWithTheme(component: React.ReactElement<any>) {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}
