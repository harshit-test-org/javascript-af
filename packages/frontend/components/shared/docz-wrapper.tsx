import React from 'react';
import { ThemeProvider } from '../../lib/styled-components';
import theme from './theme';
import { GlobalStyle } from '../Page';

const Wrapper: React.FC = props => (
  <ThemeProvider theme={theme}>
    <>
      <GlobalStyle />
      {props.children}
    </>
  </ThemeProvider>
);

export default Wrapper;
