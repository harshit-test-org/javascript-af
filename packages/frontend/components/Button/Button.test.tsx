import React from 'react';
import { Button } from './Button';
import { theme } from '../shared';
import { renderWithTheme } from '../../testUtils/renderWithTheme';
import { GithubIcon } from '../Icons/GithubIcon';

describe('Button', () => {
  test('should render primary button if no props are passed', () => {
    const { container } = renderWithTheme(<Button>Sign In</Button>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveTextContent('Sign In');
    expect(container.firstChild).toHaveStyleRule(
      'background-color',
      theme.colors.primary.main
    );
  });
  test('should render outlined variant', () => {
    const { container } = renderWithTheme(
      <Button variant="outlined">Learn More</Button>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveTextContent('Learn More');
    expect(container.firstChild).toHaveStyleRule('background-color', 'white');
    expect(container.firstChild).toHaveStyleRule(
      'border',
      `2px solid ${theme.colors.primary.main}`
    );
  });
  test('should render icon variant', () => {
    const { container } = renderWithTheme(
      <Button variant="icon">
        <GithubIcon />
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'flex');
  });
  test('should work with the as prop and other styled-system ones', () => {
    const { container } = renderWithTheme(
      <Button as="a" p={2}>
        Just test
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('padding', '8px');
    expect(container.firstChild.nodeName).toBe('A');
  });
});
