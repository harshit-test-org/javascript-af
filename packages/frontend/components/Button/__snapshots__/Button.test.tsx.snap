// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render icon variant 1`] = `
.c0 {
  border: 0;
  outline: none;
  cursor: pointer;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
  padding: 16px;
  margin: 8px;
  font-size: 1.2rem;
  border-radius: 44px;
  position: relative;
  overflow: hidden;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  padding: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:after {
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  background-image: radial-gradient(circle,#fff 10%,transparent 10.01%);
  background-repeat: no-repeat;
  background-position: 50%;
  -webkit-transform: scale(10,10);
  -ms-transform: scale(10,10);
  transform: scale(10,10);
  opacity: 0;
  -webkit-transition: -webkit-transform .5s,opacity 1s;
  -webkit-transition: transform .5s,opacity 1s;
  transition: transform .5s,opacity 1s;
}

.c0:active:after {
  -webkit-transform: scale(0,0);
  -ms-transform: scale(0,0);
  transform: scale(0,0);
  opacity: .3;
  -webkit-transition: 0s;
  transition: 0s;
}

.c0:hover {
  opacity: 0.9;
}

.c0:after {
  background-image: radial-gradient(circle,#808080 10%,transparent 10.01%);
}

<button
  class="c0"
>
  <svg
    fill="none"
    height="24"
    role="img"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
  >
    <title>
      Github icon
    </title>
    <path
      d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
    />
  </svg>
</button>
`;

exports[`Button should render outlined variant 1`] = `
.c0 {
  border: 0;
  outline: none;
  cursor: pointer;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
  padding: 16px;
  margin: 8px;
  font-size: 1.2rem;
  border-radius: 44px;
  position: relative;
  overflow: hidden;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  background-color: white;
  border-radius: 44px;
  border: 2px solid #6200EE;
  color: #6200EE;
}

.c0:after {
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  background-image: radial-gradient(circle,#fff 10%,transparent 10.01%);
  background-repeat: no-repeat;
  background-position: 50%;
  -webkit-transform: scale(10,10);
  -ms-transform: scale(10,10);
  transform: scale(10,10);
  opacity: 0;
  -webkit-transition: -webkit-transform .5s,opacity 1s;
  -webkit-transition: transform .5s,opacity 1s;
  transition: transform .5s,opacity 1s;
}

.c0:active:after {
  -webkit-transform: scale(0,0);
  -ms-transform: scale(0,0);
  transform: scale(0,0);
  opacity: .3;
  -webkit-transition: 0s;
  transition: 0s;
}

.c0:hover {
  opacity: 0.9;
}

.c0:after {
  background-image: radial-gradient(circle,#6200EE 10%,transparent 10.01%);
}

<button
  class="c0"
>
  Learn More
</button>
`;

exports[`Button should render primary button if no props are passed 1`] = `
.c0 {
  border: 0;
  outline: none;
  cursor: pointer;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
  padding: 16px;
  margin: 8px;
  font-size: 1.2rem;
  border-radius: 44px;
  position: relative;
  overflow: hidden;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  color: white;
  background-color: #6200EE;
}

.c0:after {
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  background-image: radial-gradient(circle,#fff 10%,transparent 10.01%);
  background-repeat: no-repeat;
  background-position: 50%;
  -webkit-transform: scale(10,10);
  -ms-transform: scale(10,10);
  transform: scale(10,10);
  opacity: 0;
  -webkit-transition: -webkit-transform .5s,opacity 1s;
  -webkit-transition: transform .5s,opacity 1s;
  transition: transform .5s,opacity 1s;
}

.c0:active:after {
  -webkit-transform: scale(0,0);
  -ms-transform: scale(0,0);
  transform: scale(0,0);
  opacity: .3;
  -webkit-transition: 0s;
  transition: 0s;
}

.c0:hover {
  opacity: 0.9;
}

.c0:after {
  background-image: radial-gradient(circle,#fff 10%,transparent 10.01%);
}

<button
  class="c0"
>
  Sign In
</button>
`;

exports[`Button should work with the as prop and other styled-system ones 1`] = `
.c0 {
  border: 0;
  outline: none;
  cursor: pointer;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
  padding: 16px;
  margin: 8px;
  font-size: 1.2rem;
  border-radius: 44px;
  position: relative;
  overflow: hidden;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  padding: 8px;
  color: white;
  background-color: #6200EE;
}

.c0:after {
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  background-image: radial-gradient(circle,#fff 10%,transparent 10.01%);
  background-repeat: no-repeat;
  background-position: 50%;
  -webkit-transform: scale(10,10);
  -ms-transform: scale(10,10);
  transform: scale(10,10);
  opacity: 0;
  -webkit-transition: -webkit-transform .5s,opacity 1s;
  -webkit-transition: transform .5s,opacity 1s;
  transition: transform .5s,opacity 1s;
}

.c0:active:after {
  -webkit-transform: scale(0,0);
  -ms-transform: scale(0,0);
  transform: scale(0,0);
  opacity: .3;
  -webkit-transition: 0s;
  transition: 0s;
}

.c0:hover {
  opacity: 0.9;
}

.c0:after {
  background-image: radial-gradient(circle,#fff 10%,transparent 10.01%);
}

<a
  class="c0"
>
  Just test
</a>
`;
